Library/Homebrew/utils/curl.rb

```diff
@@ -145,23 +161,19 @@ def curl(*args, print_stdout: true, **options)
       result
     end
 
-    def parse_headers(headers)
-      return {} if headers.blank?
-
-      # Skip status code
-      headers.split("\r\n")[1..].to_h do |h|
-        name, content = h.split(": ")
-        [name.downcase, content]
-      end
-    end
-
     def curl_download(*args, to: nil, try_partial: true, **options)
       destination = Pathname(to)
       destination.dirname.mkpath
 
       if try_partial
         range_stdout = curl_output("--location", "--head", *args, **options).stdout
-        headers = parse_headers(range_stdout.split("\r\n\r\n").first)
+        parsed_output = parse_curl_output(range_stdout)
+
+        headers = if parsed_output[:responses].present?
+          parsed_output[:responses].first[:headers]
```

Comment:
This is one area that may benefit from some discussion. `range_stdout.split("\r\n\r\n").first` seems to be intended to only parse the headers of the first response and ignore any others (i.e., `#parse_headers` was only able to handle one response), so I simply replicated the existing behavior in the new code.

However, this may not be appropriate for a URL that redirects, where responses before the final response would be redirections. Since the behavior in this method is determined by the presence/value of `accept-ranges` and `content-length` headers, any header differences between the first and last responses could lead to unintended behavior.

When we're downloading something, I imagine we would be primarily interested in the `accept-ranges` and `content-length` headers from the _last_ response (after any redirections). If that makes sense, it's as simple as using `parsed_output[:responses].last[:headers]` instead of `#first`.

Alternatively, we could merge the headers from all responses into one, where headers from later responses would overwrite headers from earlier responses. livecheck's `HeaderMatch` strategy does this internally (to simplify working with headers) and similar code here could look like: `parsed_output[:responses].collect { |res| res[:headers] }.reduce(&:merge)`. I would only be useful if the last response is missing headers that earlier responses include, which may not be applicable in this specific context.

I think `#last` is probably appropriate/sufficient here but I'll have to test this idea. I figured I would mention it in the interim time, to get your thoughts on this.

Reply:
> This is one area that may benefit from some discussion. `range_stdout.split("\r\n\r\n").first` seems to be intended to only parse the headers of the first response and ignore any others (i.e., `#parse_headers` was only able to handle one response), so I simply replicated the existing behavior in the new code.

Given that: can we merge this and then discuss in a future PR? I'd really like to just get this one merged out and then be able to discuss/review smaller changes.

Reply:
Works for me. I'll create a follow-up PR after I merge this in the morning (EDT).

Reply:
Shoot, I forgot that I had committed the `#last` change when I was testing it yesterday, so it was merged with the other changes earlier. I'll open a follow-up PR that would revert this one change and explain the reasoning behind why `#last` is the correct behavior in this context. If we agree on `#last`, then we can simply close the PR without merging but it will be there if we need to revert this change (highly unlikely, based on how `#curl_download` is used).

Sorry about this. I intended to incorporate this change in a follow-up PR but it was so small that I missed it in my final review.